RetailCoder.VBE/Refactorings/Rename/RenameHandlerBase.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Rubberduck.Common;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.UI;
+using Rubberduck.VBEditor;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Refactorings.Rename
+{
+    public interface IRenameRefactoringHandler
+    {
+        void Rename();
+        string ErrorMessage { get; }
+    }
+
+    public class RenameHandlerBase : IRenameRefactoringHandler
+    {
+        private readonly RenameModel _model;
+        private readonly IMessageBox _messageBox;
+        private List<QualifiedModuleName> _updatedQMNs;
+
+        public RenameHandlerBase(RenameModel model, IMessageBox messageBox)
+        {
+            _model = model;
+            _messageBox = messageBox;
+            _updatedQMNs = new List<QualifiedModuleName>();
+        }
+
+        public RenameModel Model { get { return _model; } }
+
+        public IMessageBox MessageBox { get { return _messageBox; } }
+
+        public void Rewrite()
+        {
+            _updatedQMNs.Distinct().ToList()
+                .ForEach(qmn => Model.State.GetRewriter(qmn).Rewrite());
+        }
+
+        public virtual void Rename() { }
+        public virtual string ErrorMessage { get; }
```

Comment:
A virtual member that has no sensible default shouldn't be `virtual`. Make it `abstract` if it's meant to be overridden in all derived classes.